genesyscloud/resource_genesyscloud_recording_media_retention_policy.go

```diff
@@ -0,0 +1,2715 @@
+
+		return nil
+	})
+}
+
+func createMediaRetentionPolicy(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	order := d.Get("order").(int)
+	description := d.Get("description").(string)
+	enabled := d.Get("enabled").(bool)
+	mediaPolicies := buildMediaPolicies(d)
+	conditions := buildConditions(d)
+	actions := buildPolicyActions2(d)
+	policyErrors := buildPolicyErrors(d)
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	recordingAPI := platformclientv2.NewRecordingApiWithConfig(sdkConfig)
+
+	reqBody := platformclientv2.Policycreate{
+		Name:          &name,
+		Order:         &order,
+		Description:   &description,
+		Enabled:       &enabled,
+		MediaPolicies: mediaPolicies,
+		Conditions:    conditions,
+		Actions:       actions,
+		PolicyErrors:  policyErrors,
+	}
+
+	log.Printf("Creating media retention policy %s", name)
+	// log.Printf("media retention policy POST request body %v", reqBody)
+
+	policy, apiResponse, err := recordingAPI.PostRecordingMediaretentionpolicies(reqBody)
+
+	fmt.Printf("Media retention policy creation status %#v", apiResponse.Status)
+
+	if err != nil {
+		return diag.Errorf("Failed to create media retention policy %s: %s", name, err)
+	}
+
+	// Make sure form is properly created
+	time.Sleep(2 * time.Second)
+	policyId := policy.Id
+	d.SetId(*policyId)
+	log.Printf("Created media retention policy %s %s", name, *policy.Id)
+	return readMediaRetentionPolicy(ctx, d, meta)
+}
+
+func updateMediaRetentionPolicy(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	name := d.Get("name").(string)
+	order := d.Get("order").(int)
+	description := d.Get("description").(string)
+	enabled := d.Get("enabled").(bool)
+
+	mediaPolicies := buildMediaPolicies(d)
+	conditions := buildConditions(d)
+	actions := buildPolicyActions2(d)
+	policyErrors := buildPolicyErrors(d)
+
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	recordingAPI := platformclientv2.NewRecordingApiWithConfig(sdkConfig)
+
+	reqBody := platformclientv2.Policy{
+		Name:          &name,
+		Order:         &order,
+		Description:   &description,
+		Enabled:       &enabled,
+		MediaPolicies: mediaPolicies,
+		Conditions:    conditions,
+		Actions:       actions,
+		PolicyErrors:  policyErrors,
+	}
+
+	log.Printf("Updating media retention policy %s", name)
+	policy, _, err := recordingAPI.PutRecordingMediaretentionpolicy(d.Id(), reqBody)
+	if err != nil {
+		return diag.Errorf("Failed to update media retention policy %s: %s", name, err)
+	}
+
+	log.Printf("Updated media retention policy %s %s", name, *policy.Id)
+	time.Sleep(5 * time.Second)
+	return readMediaRetentionPolicy(ctx, d, meta)
+}
+
+func mediaRetentionPolicyExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllMediaRetentionPolicies),
+		RefAttrs: map[string]*RefAttrSettings{
+			"queue_id":            {RefType: "genesyscloud_routing_queue"},
```

Comment:
These all have to refer to the full path to the id. So like something.somethingelse.queueid

Reply:
You can confirm that these are correct by exporting a policy along with users, queues, evaluation forms etc and seeing that in the generated genesyscloud/genesyscloud.tf.json file that all references have a logical reference instead of a raw ID